iothub/service/tests/ExceptionHandlingHelperTests.cs

```diff
@@ -67,28 +88,28 @@ public async Task GetExceptionCodeAsync_NoContentErrorCode_InvalidErrorCode()
             httpResponseMessage.Headers.Add(HttpErrorCodeName, "DeviceNotFound");
 
             // act
-            ErrorCode errorCode = await ExceptionHandlingHelper.GetExceptionCodeAsync(httpResponseMessage);
+            IotHubErrorCode errorCode = await ExceptionHandlingHelper.GetIotHubErrorCodeAsync(httpResponseMessage);
 
             // assert
-            Assert.AreEqual(ErrorCode.InvalidErrorCode, errorCode);
+            errorCode.Should().Be(IotHubErrorCode.Unknown);
         }
 
         [TestMethod]
-        public async Task GetExceptionCodeAsync_NoHeaderErrorCodeName_InvalidErrorCode()
+        public async Task GetExceptionCodeAsync_NoHeaderErrorCode_UnknownErrorCode()
         {
             // arrange
-            var httpResponseMessage = new HttpResponseMessage(System.Net.HttpStatusCode.BadRequest);
+            var httpResponseMessage = new HttpResponseMessage(HttpStatusCode.BadRequest);
             var exceptionResult = new IoTHubExceptionResult
             {
                 Message = "{\"errorCode\":404001}"
             };
             httpResponseMessage.Content = new StringContent(JsonConvert.SerializeObject(exceptionResult));
 
             // act
-            ErrorCode errorCode = await ExceptionHandlingHelper.GetExceptionCodeAsync(httpResponseMessage);
+            IotHubErrorCode errorCode = await ExceptionHandlingHelper.GetIotHubErrorCodeAsync(httpResponseMessage);
 
             // assert
-            Assert.AreEqual(ErrorCode.InvalidErrorCode, errorCode);
+            errorCode.Should().Be(IotHubErrorCode.Unknown);
```

Comment:
Why shouldn't this one be IotHubErrorCode.DeviceNotFound?

Reply:
We don't add `DeviceNotFound` into the response **header**, so it won't pass the check when comparing the error code from the response content to the one from response header ([ref](https://github.com/Azure/azure-iot-sdk-csharp/blob/0b950a257129594403c9b7e4e78a6499759262eb/iothub/service/src/Exceptions/ExceptionHandlingHelper.cs#L125)).

Reply:
this check is actually what we deleted near the bottom of the class "ExceptionHandlingHelper" by mistake.